pkg/antctl/raw/multicluster/join/join.go

```diff
@@ -0,0 +1,368 @@
+		return err
+	}
+
+	kubeconfig, err := raw.ResolveKubeconfig(cmd)
+	if err != nil {
+		return err
+	}
+	restConfigTmpl := rest.CopyConfig(kubeconfig)
+	raw.SetupKubeconfig(restConfigTmpl)
+
+	k8sClient, err := client.New(restConfigTmpl, client.Options{Scheme: multiclusterscheme.Scheme})
+	if err != nil {
+		return err
+	}
+
+	memberClusterNamespace := options.MemberClusterOpt.Namespace
+	memberClusterID := options.MemberClusterOpt.ClusterID
+	memberClusterSet := options.MemberClusterOpt.ClusterSet
+
+	fmt.Fprintf(cmd.OutOrStdout(), "Creating ClusterClaim \"%s\" in Namespace %s\n", multiclusterv1alpha2.WellKnownClusterClaimID, memberClusterNamespace)
+	var createErr error
+	if createErr = k8sClient.Create(context.TODO(), newClusterClaim(memberClusterID, memberClusterNamespace)); createErr != nil {
+		if apierrors.IsAlreadyExists(createErr) {
+			fmt.Fprintf(cmd.OutOrStdout(), "ClusterClaim \"%s\" already exists in Namespace %s\n", multiclusterv1alpha2.WellKnownClusterClaimID, memberClusterNamespace)
+			createErr = nil
+		} else {
+			fmt.Fprintf(cmd.OutOrStdout(), "Failed to create ClusterClaim \"%s\": %v\n", multiclusterv1alpha2.WellKnownClusterClaimID, createErr)
+			return createErr
+		}
+	} else {
+		fmt.Fprintf(cmd.OutOrStdout(), "ClusterClaim \"%s\" created in Namespace %s\n", multiclusterv1alpha2.WellKnownClusterClaimID, memberClusterNamespace)
+		defer func() {
+			if createErr != nil {
+				fmt.Fprintf(cmd.OutOrStdout(), "Deleting ClusterClaim \"%s\" in Namespace %s\n", multiclusterv1alpha2.WellKnownClusterClaimID, memberClusterNamespace)
+				if err := k8sClient.Delete(context.TODO(), newClusterClaim(memberClusterID, memberClusterNamespace)); err != nil {
+					fmt.Fprintf(cmd.OutOrStdout(), "Failed to delete ClusterClaim \"%s\": %v\n", multiclusterv1alpha2.WellKnownClusterClaimID, err)
+				} else {
+					fmt.Fprintf(cmd.OutOrStdout(), "ClusterClaim \"%s\" deleted in Namespace %s\n", multiclusterv1alpha2.WellKnownClusterClaimID, memberClusterNamespace)
+				}
+			}
+		}()
+	}
+
+	fmt.Fprintf(cmd.OutOrStdout(), "Creating ClusterClaim \"%s\" in Namespace %s\n", multiclusterv1alpha2.WellKnownClusterClaimClusterSet, memberClusterNamespace)
+	if createErr = k8sClient.Create(context.TODO(), newClusterSetClaim(options.MemberClusterOpt.ClusterSet, memberClusterNamespace)); createErr != nil {
+		if apierrors.IsAlreadyExists(createErr) {
+			fmt.Fprintf(cmd.OutOrStdout(), "ClusterClaim \"%s\" already exists in Namespace %s\n", multiclusterv1alpha2.WellKnownClusterClaimClusterSet, memberClusterNamespace)
+			createErr = nil
+		} else {
+			fmt.Fprintf(cmd.OutOrStdout(), "Failed to create ClusterClaim \"%s\": %v\n", multiclusterv1alpha2.WellKnownClusterClaimClusterSet, createErr)
+			return createErr
+		}
+	} else {
+		fmt.Fprintf(cmd.OutOrStdout(), "ClusterClaim \"%s\" created in Namespace %s\n", multiclusterv1alpha2.WellKnownClusterClaimClusterSet, memberClusterNamespace)
+		defer func() {
```

Comment:
Could you check if it's possible to merge two defer functions and move it before creation step? I feel there are too many if/else and it's not readable.

Reply:
Yes. Sorry for late reply. The outdated file is not easy to get the comments. I merged all the rollback step into one rollback function with unstructured resources.